### PR TITLE
EP-75 - iOS Obj-C SDK - Fix Code Quality Recommendations

### PR DESCRIPTION
**Pull request by stpopa**
- Code refactoring and adjustments
- OCLint issues fixed

### CODE REVIEW

File: Source/JudoKit.m [hunk 1]
```diff
@@ -73,45 +73,194 @@ - (instancetype)initWithToken:(NSString *)token secret:(NSString *)secret {
...
            completion:(nonnull void (^)(JPResponse *_Nullable, NSError *_Nullable))completion {
-
-        JudoPaymentMethodsViewModel *viewModel =
-            [[JudoPaymentMethodsViewModel alloc] initWithJudoId:judoId
-                                                         amount:amount
-                                              consumerReference:[[JPReference alloc] initWithConsumerReference:reference]
-                                                 paymentMethods:methods
-                                                    cardDetails:cardDetails];
-
+
```
Comment (zfoltin) [inline Source/JudoKit.m]: not sure these spaces are intentionally added (same on line 263)

File: Source/Model/JPCardNetwork.h [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
++ (instancetype)networkWith:(CardNetwork)type
+             numberPrefixes:(NSString *)prefixes
+         securityCodeLength:(NSUInteger)length
+              numberPattern:(NSString *)pattern;
+
++ (NSArray<JPCardNetwork *> *)supportedNetworks;
++ (NSString *)nameOfCardNetwork:(CardNetwork)network;
++ (CardNetwork)cardNetworkForCardNumber:(NSString *)cardNumber;
++ (JPCardNetwork *)cardNetworkWithType:(CardNetwork)netwrokType;
```
Comment (zfoltin) [inline Source/Model/JPCardNetwork.h]: netwrokType -> networkType

File: Source/Model/JPCardNetwork.m [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+            *stop = YES;
+            network = obj.network;
+            return;
+        }
+    }];
+    
+    return network;
+}
+
++ (JPCardNetwork *)cardNetworkWithType:(CardNetwork)netwrokType {
```
Comment (zfoltin) [inline Source/Model/JPCardNetwork.m]: netwrokType -> networkType

File: Source/Model/JPCardNetwork.m [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+#import "NSArray+Prefix.h"
+
+@implementation JPCardNetwork
+
++ (instancetype)networkWith:(CardNetwork)type
+             numberPrefixes:(NSString *)prefixes {
+    return [JPCardNetwork networkWith:type
+                       numberPrefixes:prefixes
+                   securityCodeLength:3
+                        numberPattern:kVISAPattern];
```
Comment (zfoltin) [inline Source/Model/JPCardNetwork.m]: could use `defaultNumberPattern` here?

**Comment by zfoltin** [comment]
Shit, can you wait for my review next time @stpopa especially when the PR is this big?

**Comment by stefanpopa** [comment]
> Shit, can you wait for my review next time @stpopa especially when the PR is this big?

sorry man, I thought you are not around,
we'll push requested changes in another PR

**Pull request merged**